apps/src/applab/appStorage.js

```diff
@@ -233,15 +232,14 @@ var handleDeleteRecord = function(tableName, record, onSuccess, onError) {
     return;
   }
   if (this.status === 404) {
-    onError('error deleting record: could not find record id ' + record.id +
-        ' in table ' + tableName);
+    onSuccess(false);
     return;
   }
   if (this.status < 200 || this.status >= 300) {
     onError('error deleting record: unexpected http status ' + this.status);
     return;
   }
-  onSuccess();
+  onSuccess(true);
```

Comment:
calling onComplete is mutually exclusive from calling onError. Does that answer your question?